cmd/shellNode.go

```diff
@@ -0,0 +1,210 @@
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2instanceconnect"
+
+	"github.com/instructure-bridge/truss-cli/truss"
+)
+
+var shellNodeCmd = &cobra.Command{
+	Use:   "node [node-name]",
+	Short: "Launch a shell on a Truss node.",
+	Long: `Launch a shell on a Truss node via SSH.
+
+	Usage:
+
+	truss -e [env] shell node [instance-id]
+	truss -e [env] shell node [node-name]
+
+	Examples:
+
+  truss -e nonprod-cmh shell node i-0b35c43cd48ab85ee
+	truss -e nonprod-cmh shell node ip-10-12-11-61.us-east-2.compute.internal`,
+	Args: cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		region, err := getAWSRegionFromKubeconfig()
+
+		if err != nil {
+			log.Fatal("Error retrieving AWS region from Kubeconfig:", err.Error())
+		}
+
+		awsRole := viper.GetString("aws.assumeRole")
+
+		if awsRole == "" {
+			log.Fatal("Error reading aws.assumeRole from config file:", err.Error())
+		}
+
+		sess := truss.NewAWSSession(region, awsRole)
+
+		var instanceID string
+
+		if strings.HasPrefix(args[0], "ip-") {
+			// Lookup instance-id from Kubernetes if given a node name
+			kubeNode, err := describeKubernetesNode(args[0])
+
+			if err != nil {
+				log.Fatal("Unable to describe Kubernetes node:", err.Error())
+			}
+
+			providerIDParts := strings.Split(kubeNode.Spec.ProviderID, "/")
+			instanceID = providerIDParts[len(providerIDParts)-1]
+		} else if strings.HasPrefix(args[0], "i-") {
+			instanceID = args[0]
+		}
+
+		instance := describeInstance(instanceID, sess)
+		availabilityZone := *instance.Placement.AvailabilityZone
+		hostname := *instance.PrivateDnsName
+
+		sendPublicKey(availabilityZone, instanceID, getSSHPublicKey(), sess)
+
+		jump, err := getJump()
+
+		if err != nil {
+			log.Fatal("Error encountered when finding jumpbox:", err.Error())
+		}
+
+		execSSHCommand(hostname, jump)
+
+		return nil
+	},
+}
+
+func getAWSRegionFromKubeconfig() (string, error) {
+	kubeconfig, err := getKubeconfig()
+	if err != nil {
+		return "", err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return "", err
+	}
+
+	host := config.Host
+	hostParts := strings.Split(host, ".")
+	region := hostParts[len(hostParts)-4]
+
+	return region, nil
+}
+
+func execSSHCommand(hostname string, jump string) {
+	sshBinary, err := exec.LookPath("ssh")
+	if err != nil {
+		log.Fatal("Could not find ssh binary")
+	}
+
+	target := fmt.Sprintf("ec2-user@%s", hostname)
+	proxyJump := fmt.Sprintf("ProxyJump=%s", jump)
+	syscall.Exec(sshBinary, []string{"ssh", "-o", proxyJump, target}, os.Environ())
+}
+
+func getJump() (string, error) {
+	kubeconfigName, err := getKubeconfigName()
+	if err != nil {
+		return "", err
+	}
+
+	clusterEnv := strings.Replace(kubeconfigName, "kubeconfig-truss-", "", 1)
+	jumps := viper.GetStringMap("jumps")
+
+	return jumps[clusterEnv].(string), nil
```

Comment:
why do we need to cast this to `(string)` here? Doesn't it know the values are strings because you got it with `viper.GetStringMap`?

Reply:
I changed this to do `GetString("jumps."+clusterEnv)`, which returns a string. I could have also done `GetStringMapString` and not need to cast, but I missed that possibility on my first pass. `GetStringMap` returns an interface{} which you can then either cast or unmarshal into a type.